File: app/src/main/java/com/mouredev/weeklychallenge2022/Challenge6_test.js
const reverseString = require("./Challenge6");

const tests = {
  input: ["Hola mundo", "Hello world", "here goes some text to be inverted"],
  output: ["odnum aloH", "dlrow olleH", "detrevni eb ot txet emos seog ereh"],
};

tests.input.forEach((test, index) => {
  console.log("\noriginal:", test);
  console.log(reverseString.func(test));
  console.log("expected: ", tests.output[index]);
});

File: app/src/main/java/com/mouredev/weeklychallenge2022/Challenge6_test.rb
# frozen_string_literal: true

require_relative './Challenge6'

TESTS = { input: ['Hola mundo', 'Hello world', 'here goes some text to be inverted'].freeze,
          output: ['odnum aloH', 'dlrow olleH', 'detrevni eb ot txet emos seog ereh'] }

TESTS[:input].each_with_index do |test, index|
  print "\noriginal: ", test, "\n"
  inverted_string = Main.new(test)
  puts inverted_string.execute
  puts 'expected: ', TESTS[:output][index]
end
